keps/sig-multicluster/2149-clusterid/README.md

```diff
@@ -0,0 +1,705 @@
+expanded the ways clusters can communicate with each other and organized them into `ClusterSet`s, but as of now there is no way for a cluster to be uniquely identified 
+in a Kubernetes-native way. This document by SIG-Multicluster proposes a standard 
+for how cluster IDs should be stored and managed, based on concrete use cases 
+discussed and observed in `ClusterSet` deployments. 
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+That there must be some way to identify individual clusters in a multi-cluster 
+deployment has felt like a given to SIG-Multicluster; it has been discussed in
+a broad sense previously ([see this doc](https://docs.google.com/document/d/1F__vEKeI41P7PPUCMM9PVPYY34pyrvQI5rbTJVnS5c4/edit?usp=sharing), and was scoped 
+down in response to actual observed use cases in the latest community discussion on
+which this KEP is based ([doc](https://docs.google.com/document/d/1S0u6xzP2gcJKPipA6tBNDNuid76nVKeGhTk7PrCIuQY/edit?usp=sharing)). The motivation
+of this KEP is to provide a flexible but useful baseline for clusterID that can
+work with the known use cases.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+* Propose a standard for how cluster identification metadata should be stored and 
+managed as Kubernetes resources
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+* Define any characteristics of the system that tracks cluster IDs within a cluster (i.e. a cluster registry)
+* Solve any problems without specific, tangible use cases (though we will leave room for extension).
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Overview
+Each cluster in a ClusterSet will be assigned a unique identifier that lives at least as long as its membership in the set, and is immutable for its lifetime. This identifier will be stored in a new cluster-scoped `ClusterClaim` CR with the well known name `id.k8s.io` that may be referenced by workloads within the cluster. The identifier must be a valid [RFC-1123](https://tools.ietf.org/html/rfc1123) DNS label, and may be created by an implementation dependent mechanism.
```

Comment:
> for its lifetime

Whose lifetime?  The cluster?  Or the CR?  If the CR, are _all_ ClusterClaims immutable?  Or just this named one?  What happens when it is, inevitably, mutated by someone (deleted and recreated, for example)

Reply:
I think this needs to be `for the lifetime of a cluster's membership in a clusterset`. I don't think we can make uniqueness requirements with infinite scope, and tying to membership seems enough to make it useful.

Reply:
Cleaned up the language in 75cab4d9e1bdc20f9bbd1cd086041d4a96103abb, including adding the remove/adding cluster scenario explicitly